plugins/net_plugin/net_plugin.cpp

```diff
@@ -254,8 +266,9 @@ namespace eosio {
       bool                                  use_socket_read_watermark = false;
       /** @} */
 
-      mutable std::shared_mutex             connections_mtx;
+      mutable std::shared_mutex             connections_mtx; // protects both connections and security_group
```

Comment:
Why share this mutex with security_group? Seems like the code would be much simplier if security_group_manger had its own mutex to protect its internal data.

Reply:
that was my fault, I was thinking that it was going to be held while adding a connection on the TLS side, so I was trying to avoid the issues of double locking. Will change.